src/types/boxed.rs

```diff
@@ -0,0 +1,343 @@
+use std::any::{self, Any};
+use std::ops::Deref;
+
+use neon_runtime::raw;
+use neon_runtime::external;
+
+use crate::context::{Context, FinalizeContext};
+use crate::context::internal::Env;
+use crate::handle::{Managed, Handle};
+use crate::types::internal::ValueInternal;
+use crate::types::Value;
+
+type BoxAny = Box<dyn Any + Send + 'static>;
+
+/// A smart pointer for Rust data managed by the JavaScript engine.
+///
+/// The type `JsBox<T>` provides shared ownership of a value of type `T`,
+/// allocated in the heap. The data is owned by the JavaScript engine and the
+/// lifetime is managed by the JavaScript garbage collector.
+///
+/// Shared references in Rust disallow mutation by default, and `JsBox` is no
+/// exception: you cannot generally obtain a mutable reference to something
+/// inside a `JsBox`. If you need to mutate through a `JsBox`, use
+/// [`Cell`](https://doc.rust-lang.org/std/cell/struct.Cell.html),
+/// [`RefCell`](https://doc.rust-lang.org/stable/std/cell/struct.RefCell.html),
+/// or one of the other types that provide interior mutability.
```

Comment:
Let's link the phrase "interior mutability" to the [book chapter on interior mutability](https://doc.rust-lang.org/book/ch15-05-interior-mutability.html).